File: backend/hotelEase/hotelEase.Model/Hotel.cs
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Text;

namespace hotelEase.Model
{
    public class Hotel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public string? Description { get; set; }

        public string Address { get; set; } = null!;

        public int CityId { get; set; }

        public int CountryId { get; set; }

        public int ManagerId { get; set; }

        public int StarRating { get; set; }

        public bool? IsActive { get; set; }
        public bool? SPA { get; set; }
        public bool? Parking { get; set; }
        public bool? WiFi { get; set; }
        public bool? Pool { get; set; }
        public bool? Bar { get; set; }
        public bool? Fitness { get; set; }

        public DateTime? CreatedAt { get; set; }
        public string? StateMachine { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? DeletedTime { get; set; }

        public virtual ICollection<Room> Rooms { get; set; } = new List<Room>();
        public virtual City City { get; set; } = null!;

        public decimal MinPrice { get; set; }
        public decimal MaxPrice { get; set; }

    }
}

File: UI/hotelease_mobile_new/lib/models/hotel.dart
import 'package:hotelease_mobile_new/models/asset.dart';
import 'package:hotelease_mobile_new/models/city.dart';
import 'package:hotelease_mobile_new/models/room.dart';
import 'package:json_annotation/json_annotation.dart';

part 'hotel.g.dart';

@JsonSerializable()
class Hotel {
  int? id;
  String? name;
  int? price;
  String? description;
  String? address;

  int? cityId;

  int? starRating;
  List<Asset>? assets;
  List<Room>? rooms;

  bool? wifi;
  bool? parking;
  bool? pool;
  bool? bar;
  bool? fitness;
  bool? spa;
  String? imageUrl;

  int? minPrice;
  int? maxPrice;
  City? city;
  Hotel({
    this.imageUrl,
    this.id,
    this.name,
    this.price,
    this.description,
    this.address,
    this.cityId,

    this.starRating,
    this.assets,
    this.rooms,
    this.bar,
    this.fitness,
    this.parking,
    this.pool,
    this.spa,
    this.wifi,

    this.minPrice,
    this.maxPrice,
    this.city,
  });

  factory Hotel.fromJson(Map<String, dynamic> json) => _$HotelFromJson(json);

  Map<String, dynamic> toJson() => _$HotelToJson(this);
}
